Source/Scene/Cesium3DTilesetAsyncTraversal.js

```diff
@@ -0,0 +1,141 @@
+define([
+        '../Core/Intersect',
+        '../Core/ManagedArray',
+        './Cesium3DTileRefine'
+    ], function(
+        Intersect,
+        ManagedArray,
+        Cesium3DTileRefine) {
+    'use strict';
+
+    /**
+     * Traversal that loads all leaves that intersect the camera frustum.
+     * Used to determine ray-tileset intersections during a pickFromRayMostDetailed call.
+     *
+     * @private
+     */
+    function Cesium3DTilesetAsyncTraversal() {
+    }
+
+    var asyncTraversal = {
+        stack : new ManagedArray(),
+        stackMaximumLength : 0
+    };
+
+    Cesium3DTilesetAsyncTraversal.selectTiles = function(tileset, frameState) {
+        tileset._selectedTiles.length = 0;
+        tileset._requestedTiles.length = 0;
+        tileset._hasMixedContent = false;
+        var minimumGeometricError = 0;
+
+        var ready = true;
+
+        var root = tileset.root;
+        root.updateVisibility(frameState);
+
+        if (!isVisible(root)) {
+            return ready;
+        }
+
+        if (tileset._geometricError <= minimumGeometricError) {
+            return ready;
+        }
+
+        var stack = asyncTraversal.stack;
+        stack.push(tileset.root);
+
+        while (stack.length > 0) {
+            asyncTraversal.stackMaximumLength = Math.max(asyncTraversal.stackMaximumLength, stack.length);
+
+            var tile = stack.pop();
+            var add = tile.refine === Cesium3DTileRefine.ADD;
```

Comment:
Here and below, wrap the righthand side of the expression in parentheses.  Please add that to the Coding Guide if it isn't already.

Reply:
It is already in the guide. Fixed.